models/m_payment_mollie_api.erl

```diff
@@ -448,95 +480,154 @@ api_key(Context) ->
         } ->
-            case api_call(get, "customers/" ++ z_convert:to_list(CustomerId) ++ "/mandates", [], Context) of
-                {ok, Response} ->
-                    Mandates = maps:get(<<"data">>, Response, []),
-                    case lists:filter(fun valid_mandate/1, Mandates) of
-                        [] ->
+            % v1 API data
+            create_subscription_1(FirstPayment, CustomerId, Context);
+        PspData ->
+            % Log an error with the payment
+            {id, PaymentId} = proplists:lookup(id, FirstPayment),
+            {user_id, UserId} = proplists:lookup(user_id, FirstPayment),
+            m_payment_log:log(
+                PaymentId,
+                <<"ERROR">>,
+                [
+                    {psp_module, mod_payment_mollie},
+                    {description, <<"Could not create subscription: PSP data missing sequenceType and/or customerId">>}
+                ],
+                Context),
+            lager:error("Mollie payment PSP data missing sequenceType and/or customerId, payment ~p (user ~p): ~p",
+                        [ PaymentId, UserId, PspData ]),
+            {error, pspdata}
+    end.
+
+create_subscription_1(FirstPayment, CustomerId, Context) ->
+    {id, PaymentId} = proplists:lookup(id, FirstPayment),
+    {user_id, UserId} = proplists:lookup(user_id, FirstPayment),
+    case api_call(get, "customers/" ++ z_convert:to_list(CustomerId) ++ "/mandates", [], Context) of
+        {ok, #{
+            <<"_embedded">> := #{
+                <<"mandates">> := Mandates
+            }
+        }} ->
+            case lists:any(fun is_valid_mandate/1, Mandates) of
+                true ->
+                    {{Year, Month, Day}, _} = calendar:local_time(),
+                    YearFromNow = io_lib:format("~4..0w-~2..0w-~2..0w", [Year + 1, Month, Day]),
+                    Email = z_convert:to_binary( m_rsc:p_no_acl(UserId, email, Context) ),
+                    WebhookUrl = webhook_url(proplists:get_value(payment_nr, FirstPayment), Context),
+                    Args = [
+                        {'amount[value]', filter_format_price:format_price(proplists:get_value(amount, FirstPayment), Context)},
+                        {'amount[currency]', proplists:get_value(currency, FirstPayment, <<"EUR">>)},
+                        {interval, <<"12 months">>},
+                        {startDate, list_to_binary(YearFromNow)},
+                        {webhookUrl, WebhookUrl},
+                        {description, <<"Subscription for ", Email/binary, " - ", CustomerId/binary>>}
+                    ],
+                    case api_call(post, "customers/" ++ z_convert:to_list(CustomerId) ++ "/subscriptions",
+                                  Args, Context)
+                    of
+                        {ok, #{
+                            <<"resource">> := <<"subscription">>,
+                            <<"status">> := SubStatus
+                        } = Sub} ->
```

Comment:
Does `/subscriptions` return a single subscription?

Reply:
We POST to the API to create a subscription, and then the new subscription is returned.

https://docs.mollie.com/reference/create-subscription